lib/errgonomic.rb

```diff
@@ -43,14 +51,39 @@ class ResultRequiredError < Error; end
   module Result
     class Any
 
+      # Equality comparison for Result objects is based on value not reference.
+      #
+      # @example
+      #   Ok("foo") == Ok("foo") # => true
+      #   Ok("foo") == Err("foo") # => false
+      #   Ok("foo").object_id != Ok("foo").object_id # => true
+      def ==(other)
+        self.class == other.class && self.value == other.value
+      end
+
       # Indicate that this is some kind of result object. Contrast to
       # Object#result? which is false for all other types.
+      # @example
+      #   Ok("foo").result? # => true
+      #   Err("foo").result? # => true
+      #   "foo".result? # => false
       def result?
         true
       end
 
       # A lightweight DSL to invoke code for a result based on whether it is an
       # Ok or Err.
+      #
+      # @example
+      #   Ok("foo").match do
+      #     ok { :foo }
+      #     err { :bar }
+      #   end # => :foo
+      #
+      #   Err("foo").match do
+      #     ok { :foo }
+      #     err { :bar }
+      #   end # => :bar
```

Comment:
1) Love the doc tests
2) I see Rust's match as requiring that all cases are handled (exhaustive pattern matching) and also to expose inner values. 

Based on this example, it's unclear if it yields or gives the original value somehow. It's not quite [and_then](https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then) but that is what this reminds me of. Actually maybe it's closer to https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or_else.

3) Random unsolicited advice, this cheatsheet is really helpful https://upsuper.github.io/rust-cheatsheet/?dark,large

Reply:
Yeah I borrowed this from another random gem that had taken a stab at Rust style Results, and I don't love it. I'm going to work up a proper example with case and probably ditch this unless I can come up with a better reason to have it.